src/pytorch_lightning/core/module.py

```diff
@@ -440,7 +446,7 @@ def log(
             logger=logger,
             on_step=on_step,
             on_epoch=on_epoch,
-            reduce_fx=reduce_fx,
+            reduce_fx=reduce_fx,  # type: ignore[arg-type]
```

Comment:
the implementation of `module.log` (ln319) allows the user to pass in a str or Callable for arg `reduce_fx`; overriding the _Metadata dataclass in `pl.trainer.connectors.logger_connector`, which uses `reduce_fx: Callable = torch.mean`.

updating _Metadata type hints did not resolve the issue - it caused more. 

 an alternative to ignoring the error would be to use `pl.trainer.connectors.logger_connector._Metadata._parse_reduce_fx` in `pl.core.module` before calling `results.log` at ln 441.